com.unity.multiplayer.mlapi/Runtime/AOI/ClientObjectMap.cs

```diff
@@ -0,0 +1,113 @@
+// Although this is (currently) inside the MLAPI package, it is intentionally
+//  totally decoupled from MLAPI with the intention of allowing it to live
+//  in its own package
+namespace MLAPI.AOI
+{
+    // To establish a Client Object Map, instantiate a ClientObjMapNodeBase, then
+    //  add more nodes to it (and those nodes) as desired
+   public class ClientObjMapNode<CLIENT, OBJECT>
```

Comment:
Can object really be anything? Or should it be constrained ?

Reply:
Yeah, at least how I have it implemented it doesn't care.  The delegates you attach can care, as can subclasses, but the base class - not yet.  As there was interest in AOI outside of MLAPI I tried to keep it generic.  What were you thinking?

Reply:
I'd like to suggest calling template params as `NetClient` and `NetObject` (not all in caps, still avoids Object vs System.Object collision and "Net" prefix should make it a little bit more indicative too) — another alternative might be `NetworkClient`/`NetworkObject` pair to be even more explicit about our intentions and names.

Reply:
Chatting with @wackoisgod , he shared the very good suggestion to make sure that we constrain to ref types (vs. value types), which I can do with the ref keyword

Reply:
I'll go with JetBrain's suggestion, which is CLIENT => TClient, OBJECT => TObject